test/index.js

```diff
@@ -88,35 +111,28 @@ if (napiVersion < 6) {
   testModules.splice(testModules.indexOf('addon_data'), 1);
 }
 
-if (typeof global.gc === 'function') {
-  console.log(`Testing with N-API Version '${napiVersion}'.`);
-
-  console.log('Starting test suite\n');
-
-  // Requiring each module runs tests in the module.
-  testModules.forEach(name => {
-    console.log(`Running test '${name}'`);
-    require('./' + name);
-  });
-
-  console.log('\nAll tests passed!');
-} else {
-  // Construct the correct (version-dependent) command-line args.
-  let args = ['--expose-gc', '--no-concurrent-array-buffer-freeing'];
-  if (majorNodeVersion >= 14) {
-    args.push('--no-concurrent-array-buffer-sweeping');
-  }
-  args.push(__filename);
-
-  const child = require('./napi_child').spawnSync(process.argv[0], args, {
-    stdio: 'inherit',
-  });
+console.log(`Testing with N-API Version '${napiVersion}'.`);
+console.log('Starting test suite\n');
 
+function runOneChild(name, mode) {
+  console.log(`Running test '${name}' as ${mode}`);
+  const child = require('./napi_child').spawnSync(process.execPath, [
+    __filename, name, mode
+  ], { stdio: 'inherit' });
   if (child.signal) {
-    console.error(`Tests aborted with ${child.signal}`);
-    process.exitCode = 1;
-  } else {
+    console.error(`Test ${name} run as ${mode} aborted with ${child.signal}`);
+    if (!process.exitCode) {
+      process.exitCode = 1;
+    }
+  }
+  if (!process.exitCode && child.status !== 0) {
     process.exitCode = child.status;
   }
-  process.exit(process.exitCode);
 }
+
+testModules.forEach((name) => {
+  runOneChild(name, 'child');
```

Comment:
I'd worried that all the cases were started at the same time. Can we wait for one case to finish then start another one?